e2etest/bgp_test.go

```diff
@@ -267,20 +267,38 @@ var _ = ginkgo.Describe("BGP", func() {
 			}
 
 			ginkgo.By("checking the metrics when no service is added")
-			controllerMetrics, err := metrics.ForPod(controllerPod, controllerPod)
-			framework.ExpectNoError(err)
-			validateGaugeValue(0, "metallb_allocator_addresses_in_use_total", map[string]string{"pool": poolName}, controllerMetrics)
-			validateGaugeValue(272, "metallb_allocator_addresses_total", map[string]string{"pool": poolName}, controllerMetrics)
+			Eventually(func() error {
+				controllerMetrics, err := metrics.ForPod(controllerPod, controllerPod)
+				framework.ExpectNoError(err)
```

Comment:
why not return the err (also applies to the other calls to metrics.ForPod)?

Reply:
Because not being able to fetch the metrics should not be depending on timing. The content is, and that's why we added the eventually loop

Reply:
Done